Fixtures/MixedTargets/BasicMixedTargets/Package.swift

```diff
@@ -0,0 +1,271 @@
+        // This target is an edge case. It's purpose may or may not be
+        // useful, but it shouldn't fail to build.
+       .target(
+           name: "MixedTargetWithNoPublicObjectiveCHeaders"
+       ),
+        .testTarget(
+            name: "MixedTargetWithNoPublicObjectiveCHeadersTests",
+            dependencies: ["MixedTargetWithNoPublicObjectiveCHeaders"]
+        ),
+
+        // MARK: - MixedTargetWithNoObjectiveCCompatibleSwiftAPI
+        .target(
+           name: "MixedTargetWithNoObjectiveCCompatibleSwiftAPI"
+       ),
+
+        // MARK: - MixedTestTargetWithSharedTestUtilities
+        .testTarget(
+            name: "MixedTestTargetWithSharedUtilitiesTests"
+        ),
+
+        // MARK: - PrivateHeadersCanBeTestedViaHeaderSearchPathsTests
+        .testTarget(
+            name: "PrivateHeadersCanBeTestedViaHeaderSearchPathsTests",
+            dependencies: ["MixedTargetWithNonPublicHeaders"],
+            cSettings: [
+                // Adding a header search path at the root of the package will
+                // enable the Objective-C tests to import private headers.
+                .headerSearchPath("../../")
+            ]
+        ),
+
+        // MARK: - MixedTargetsPublicHeadersAreIncludedInHeaderSearchPathsForObjcSource
+        .target(
+            name: "MixedTargetsPublicHeadersAreIncludedInHeaderSearchPathsForObjcSource"
+        ),
+
+        // MARK: - Targets for testing the integration of a mixed target
+        .target(
+            name: "ClangTargetDependsOnMixedTarget",
+            dependencies: ["BasicMixedTarget"]
+         ),
+         .target(
+             name: "SwiftTargetDependsOnMixedTarget",
+             dependencies: ["BasicMixedTarget"]
+         ),
+         .target(
+             name: "MixedTargetDependsOnMixedTarget",
+             dependencies: ["BasicMixedTarget"]
+         ),
+         .target(
+             name: "MixedTargetDependsOnClangTarget",
+             dependencies: ["ClangTarget"]
+         ),
+         .target(
+             name: "MixedTargetDependsOnSwiftTarget",
+             dependencies: ["SwiftTarget"]
+         ),
+         // The below two targets are used for testing the above targets.
+         .target(name: "SwiftTarget"),
+         .target(name: "ClangTarget")
+    ] + targetsWithCxxInteropEnabled(),
+    // TODO(ncooke3): Is the below note behavior that we want to be intended?
+    // This is needed for targets with that have
+    // `swiftSettings: [.interoperabilityMode(.Cxx)]` set.
+    cxxLanguageStandard: .cxx11
+)
+
+func targetsWithCxxInteropEnabled() -> [Target] {
+// The below targets have C++ interoperability mode enabled, a feature that
+// requires Swift 5.9 or greater.
+#if swift(>=5.9)
+  // MARK: - MixedTargetWithCXX_CXXInteropEnabled
+  return [
+    .target(
+      name: "MixedTargetWithCXX_CXXInteropEnabled",
+      swiftSettings: [.interoperabilityMode(.Cxx)]
+    )
+  ]
+#else
+  return []
+#endif  // swift(>=5.9)
```

Comment:
Hi @hyp, does it seem expected that a package that contains targets opting into C++ interop _must_ specify a `cxxLanguageStandard` at the package level?

This code is from a `Package.swift` containing mixed targets that are tested in the test suite. When I specify a `cxxLanguageStandard` (line 254), building the `MixedTargetWithCXX_CXXInteropEnabled` target is successful.  When I don't (comment out line  254), the build fails with the below error. It looks like the compiler doesn't realize the Swift header uses C++ semantics or something. 

My hunch is this should work without needing to setting the `cxxLanguageStandard`. I'm thinking this since the example package definition at the top of these [docs](https://www.swift.org/documentation/cxx-interop/project-build-setup/#mixing-swift-and-c-using-swift-package-manager) does not specific a language standard.

This seems semi-related to work done in https://github.com/apple/swift-package-manager/pull/6652.

<details><summary>Expand for error message...</summary>
<p>

```
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled/SwiftCountdown.swift -emit-dependencies-path /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/SwiftCountdown.d -emit-reference-dependencies-path /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/SwiftCountdown.swiftdeps -target arm64-apple-macosx10.13 -Xllvm -aarch64-use-tbi -enable-objc-interop -cxx-interoperability-mode=default -sdk /Applications/Xcode_15_b2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -I /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug -I /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build -I /Applications/Xcode_15_b2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib -F /Applications/Xcode_15_b2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -color-diagnostics -enable-testing -g -import-underlying-module -module-cache-path /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/ModuleCache -swift-version 5 -Onone -D SWIFT_PACKAGE -D DEBUG -new-driver-path /Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-driver -empty-abi-descriptor -plugin-path /Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/plugins -plugin-path /Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/local/lib/swift/host/plugins -resource-dir /Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -ivfsoverlay -Xcc /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/unextended-module-overlay.yaml -Xcc -I -Xcc /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled -Xcc -I -Xcc /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled/include -Xcc -isysroot -Xcc /Applications/Xcode_15_b2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -Xcc -F -Xcc /Applications/Xcode_15_b2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -Xcc -fPIC -Xcc -g -module-name MixedTargetWithCXX_CXXInteropEnabled -package-name basicmixedtargets -disable-clang-spi -target-sdk-version 14.0 -target-sdk-name macosx14.0 -parse-as-library -o /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/SwiftCountdown.swift.o -index-store-path /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/index/store -index-system-modules
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -fobjc-arc -target arm64-apple-macosx10.13 -O0 -DSWIFT_PACKAGE=1 -DDEBUG=1 -fblocks -index-store-path /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/index/store -I /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled/include -I /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled -I /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build -isysroot /Applications/Xcode_15_b2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -F /Applications/Xcode_15_b2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -fPIC -g -g -MD -MT dependencies -MF /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/CxxCountdown.cpp.d -c /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled/CxxCountdown.cpp -o /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/CxxCountdown.cpp.o
In file included from /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled/CxxCountdown.cpp:4:
In file included from /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:308:
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:92:56: error: expected function body after function declarator
extern "C" void *_Nonnull swift_retain(void *_Nonnull) noexcept;
                                                       ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:94:47: error: expected function body after function declarator
extern "C" void swift_release(void *_Nonnull) noexcept;
                                              ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:103:64: error: expected function body after function declarator
                                               uint32_t flags) noexcept;
                                                               ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:112:61: error: expected function body after function declarator
                                              size_t align) noexcept {
                                                            ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:173:67: error: expected ';' at end of declaration list
  SWIFT_INLINE_THUNK RefCountedClass(const RefCountedClass &other) noexcept
                                                                  ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:172:57: error: use of undeclared identifier '_opaquePointer'
  SWIFT_INLINE_THUNK ~RefCountedClass() { swift_release(_opaquePointer); }
                                                        ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:213:19: error: no member named '_opaquePointer' in 'swift::_impl::RefCountedClass'
    return object._opaquePointer;
           ~~~~~~ ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:217:19: error: no member named '_opaquePointer' in 'swift::_impl::RefCountedClass'
    return object._opaquePointer;
           ~~~~~~ ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:221:25: error: no member named '_opaquePointer' in 'swift::_impl::RefCountedClass'
    swift_retain(object._opaquePointer);
                 ~~~~~~ ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:222:19: error: no member named '_opaquePointer' in 'swift::_impl::RefCountedClass'
    return object._opaquePointer;
           ~~~~~~ ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:240:21: error: unknown type name 'constexpr'
static inline const constexpr bool isUsableInGenericContext = false;
                    ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:256:40: error: unknown type name 'constexpr'
template <class T> static inline const constexpr bool isValueType = false;
                                       ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:260:40: error: unknown type name 'constexpr'
template <class T> static inline const constexpr bool isOpaqueLayout = false;
                                       ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:265:21: error: unknown type name 'constexpr'
static inline const constexpr bool isSwiftBridgedCxxRecord = false;
                    ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:270:6: error: expected '(' after 'if'
  if constexpr (isOpaqueLayout<T>)
     ^
/Applications/Xcode_15_b2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/../../../swift/swiftToCxx/_SwiftCxxInteroperability.h:277:6: error: expected '(' after 'if'
  if constexpr (isOpaqueLayout<T>)
     ^
In file included from /Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/Sources/MixedTargetWithCXX_CXXInteropEnabled/CxxCountdown.cpp:4:
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:359:56: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessInitializeBufferWithCopyOfBufferTy = void * _Nonnull(* __ptrauth_swift_value_witness_function_pointer(55882))(void * _Nonnull, void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                                       ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:360:31: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessDestroyTy = void(* __ptrauth_swift_value_witness_function_pointer(1272))(void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                              ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:361:42: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessInitializeWithCopyTy = void * _Nonnull(* __ptrauth_swift_value_witness_function_pointer(58298))(void * _Nonnull, void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                         ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:362:38: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessAssignWithCopyTy = void * _Nonnull(* __ptrauth_swift_value_witness_function_pointer(34641))(void * _Nonnull, void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                     ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:363:42: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessInitializeWithTakeTy = void * _Nonnull(* __ptrauth_swift_value_witness_function_pointer(18648))(void * _Nonnull, void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                         ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:364:38: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessAssignWithTakeTy = void * _Nonnull(* __ptrauth_swift_value_witness_function_pointer(61402))(void * _Nonnull, void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                     ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:365:47: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessGetEnumTagSinglePayloadTy = unsigned(* __ptrauth_swift_value_witness_function_pointer(24816))(const void * _Nonnull, unsigned, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                              ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:366:49: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using ValueWitnessStoreEnumTagSinglePayloadTy = void(* __ptrauth_swift_value_witness_function_pointer(41169))(void * _Nonnull, unsigned, unsigned, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                                ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:382:3: error: unknown type name 'constexpr'
  constexpr size_t getAlignment() const { return (flags & 255) + 1; }
  ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:382:19: error: expected ';' at end of declaration list
  constexpr size_t getAlignment() const { return (flags & 255) + 1; }
                  ^
                  ;
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:385:38: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using EnumValueWitnessGetEnumTagTy = unsigned(* __ptrauth_swift_value_witness_function_pointer(41909))(const void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                     ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:386:54: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using EnumValueWitnessDestructiveProjectEnumDataTy = void(* __ptrauth_swift_value_witness_function_pointer(1053))(void * _Nonnull, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                                     ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:387:52: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
using EnumValueWitnessDestructiveInjectEnumTagTy = void(* __ptrauth_swift_value_witness_function_pointer(45796))(void * _Nonnull, unsigned, void * _Nonnull) SWIFT_NOEXCEPT_FUNCTION_PTR;
                                                   ^
/Users/nickcooke/Developer/swift-package-manager/Fixtures/MixedTargets/BasicMixedTargets/.build/arm64-apple-macosx/debug/MixedTargetWithCXX_CXXInteropEnabled.build/MixedTargetWithCXX_CXXInteropEnabled-Swift.h:435:21: error: unknown type name 'constexpr'
static inline const constexpr bool isUsableInGenericContext<bool> = true;
                    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
11 warnings and 20 errors generated.
Program ended with exit code: 1
```

</p>
</details>

Reply:
There is another similar-ish error related to this target. It happens when building on Linux and was surfaced on CI: https://ci.swift.org/job/swift-package-manager-self-hosted-Linux-smoke-test/2681/console 

It seems that in this other case when building on Linux, setting the `cxxLanguageStandard` is not enough, and the generated `-Swift.h` header does not build. I suspect this is related to me removing the constraint where the -Swift.h header is emitted only on DarwinOS, but I haven't got to the bottom of it yet. https://github.com/apple/swift-package-manager/pull/5919/commits/3202a825c8afcd527e3d808c3be5c2a3c047f263#diff-31b280ef997c175a0c39198547ce45d473e9fb89ac71aaa5ed2ad7e4a92473c8L612

Reply:
The linux error is weird, it looks like you're mixing toolchains and one has outdated base header?

Reply:
> Hi @hyp, does it seem expected that a package that contains targets opting into C++ interop must specify a cxxLanguageStandard at the package level?
Not really. Can you post the full build log with the error you're seeing instead of just the snippet?

Reply:
> > Hi @hyp, does it seem expected that a package that contains targets opting into C++ interop must specify a cxxLanguageStandard at the package level?
> 
> Not really. Can you post the full build log with the error you're seeing instead of just the snippet?

Here is a gist containing the build logs from running `swift build` on the target with the `--very-verbose` flag. https://gist.github.com/ncooke3/1f600bf2b9c7404fea44d489d0797c80

Reply:
> The linux error is weird, it looks like you're mixing toolchains and one has outdated base header?

Hi @hyp, the Linux error seems to have resolved itself. But, I'm still facing the original error in the above gist.

Reply:
The original error happens because Apple clang defaults to C++98 still, and the generated header supports C++14 and above only. Unfortunately this is still a limitation that needs to be resolved, but not sure how.